sling/nlp/wikicat/workflow.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Workflow for generating and scoring parses for Wikipedia category titles.
+
+import sling
+import sling.flags as flags
+import sling.task.corpora as corpora
+from sling.task.workflow import Workflow, start_monitor
+
+# Workflow tasks.
+import generator
+import prelim_ranker
+
+
+class CategoryParsingWorkflow:
+  def __init__(self, name, outdir, wf=None):
+    if wf is None: wf = Workflow(name)
+    self.wf = wf
+    self.outdir = outdir
+
+
+  # Adds a KB input to 'task'.
+  def kb_input(self, task, kb_dir=None):
+    if kb_dir is None:
+      kb_dir = corpora.wikidir()
+    kb = self.wf.resource(file="kb.sling", dir=kb_dir, format="store/frame")
+    task.attach_input("kb", kb)
+
+
+  # Stage 1: Generate all parses.
+  def generate_parses(self, language, min_members):
+    with self.wf.namespace("generate-parses"):
+      generator = self.wf.task("parse-generator")
+      generator.add_params({
+        "language": language,
+        "min_members": min_members
+      })
+      wikidir = corpora.wikidir()
+      self.kb_input(generator, kb_dir=wikidir)
+
+      members = self.wf.resource(
+          file="wikipedia-members.rec", dir=wikidir, format="records/frame")
+      generator.attach_input("members", members)
+
+      phrase_table = self.wf.resource(
+          "phrase-table.repo", dir=wikidir + "/en", format="text/frame")
```

Comment:
You should use the language argument for locating the phrase table for the language.

Reply:
Good catch, fixed.